integreat_cms/static/src/js/pages/collapse-subpages.ts

```diff
@@ -65,8 +91,91 @@ function toggleSubpagesRecursive(collapseSpan: HTMLElement) {
-        toggleSubpagesRecursive(collapseSpan);
+        const grandChildren: number[] = JSON.parse(
+          collapseSpan.getAttribute("data-page-children")
+        );
+        toggleSubpagesRecursive(grandChildren);
       }
     }
   });
 }
+
+/**
+ * Ajax call to fetch children of selected page 
+ * and insert them into DOM at right position
+ * 
+ * @param collapseSpan The page to expand its children
+ * @returns Array of children ids
+ */
+async function fetchSubpages(collapseSpan: HTMLElement): Promise<number[]> {
+  // Get children of page
+  let response = await fetch(collapseSpan.dataset.pageChildrenUrl);
+  if (!response.ok) {
+    throw new Error("Failed with status code: " + response.status);
+  }
+  let html = await response.text();
+  var childrenIds: number[] = [];
+  // Convert the HTML string into a document object
+	let parser = new DOMParser();
+	let childrenDoc = parser.parseFromString(html, 'text/html');
+  if (childrenDoc.querySelectorAll("tr").length === 0) {
+    return childrenIds;
+  }
+  let parentRow = collapseSpan.closest("tr");
+  childrenDoc.querySelectorAll("tr").forEach(function(rowToinsert: HTMLTableRowElement) {
+    let newCollapseSpan = rowToinsert.querySelector(".collapse-subpages");
+    if (newCollapseSpan) {
+      // add event listener for page expansion if not leaf node
+      newCollapseSpan.addEventListener("click", toggleSubpages);
+    }
+    if (!rowToinsert.classList.contains("drop-between")) {
+      // record children ids to update all ancestors descendants data attribute
+      childrenIds.push(parseInt(rowToinsert.dataset.dropId));
+    }
+    // insert child into DOM tree
+    parentRow.parentNode.insertBefore(rowToinsert, parentRow.nextSibling);
+    // update point to insert next child
+    parentRow = rowToinsert;
+  });
+  updateDescendants(childrenDoc, childrenIds);
+  updateChildren(collapseSpan, childrenIds);
+  // render icons for recently added DOM elements
+  feather.replace({ class: 'inline-block' });
+  return childrenIds;
+}
+
+/**
+ * Adds new descendants to data-node-descendants of all ancestor
+ * 
+ * @param childrenDoc document containing ancestor information
+ * @param childrenIds the chidlren ids to be added
+ */
+function updateDescendants(childrenDoc: Document, childrenIds: Array<number>) {
+  // get ids of all ancestors for the selected page
+  let ancestors = JSON.parse(
+    (<HTMLElement>childrenDoc.querySelector("#ancestors")).dataset.ancestors
+  ) as number[];
+  ancestors.forEach(function(ancestor: number) {
+    // as new children were added to the DOM, all ancestors have to update their descendants 
+    let ancestorDragSpan = <HTMLElement>document.querySelector("#page-" + ancestor + " .drag");
+    const descendants = JSON.parse(
+      ancestorDragSpan.dataset.nodeDescendants
+    ) as number[];
+    // update descendants
+    ancestorDragSpan.dataset.nodeDescendants = "[" + descendants.concat(childrenIds).join(",") + "]";
```

Comment:
```suggestion
    ancestorDragSpan.dataset.nodeDescendants = `[${descendants.concat(childrenIds)}]`;
```
Lists are automatically joined with commas when converted to a string.